lib/pbench/agent/tool_meister.py

```diff
@@ -467,6 +467,39 @@ def install(self):
             return (1, "pcp tool (pmcd) not found")
         return (0, "pcp tool (pmcd) properly installed")
 
+class JaegerTool(PersistentTool):
+    """JaegerTool - provides specifics for running a Jaeger Agent on the host"""
+
+    def __init__(self, name, tool_opts, logger=None, **kwargs):
+        super().__init__(name, tool_opts, logger=logger, **kwargs)
+
+        # get the collector host_post for tool options
+        self.collector_hosts_ports = None
+        for opt in self.tool_opts:
+            if opt.startswith("--collector="):
+                if opt[len(opt) - 1] == "\n":
+                    self.collector_hosts_ports = opt[12 : len(opt) - 1]
+                else:
+                    self.collector_hosts_ports = opt[12:]
+        if self.collector_hosts_ports is None:
+            self.logger.error("No host:port provided for jaeger collector(s)")
+
+        executable = find_executable("jaeger-agent")
+        if executable:
+            self.args = [
+                executable,
+                ("--reporter.grpc.host-port=%s" % self.collector_hosts_ports),
```

Comment:
Consider dropping `--collector` option to simplify, and use the "tool data sink" host name (which we have to find the right field name) and a default port, using Python3 f-strings: `f"--reporter.grpc.host-port={self.tds_host_name}:{self.def_jaeger_port}"`